src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -182,10 +229,10 @@ const TopicFragment = ({
         unfollowingTopics.push(item);
       }
     });
-
+    const firstData = isFirstTimeOpen ? followingTopics : newMapFollowedTopics;
     const data = isFirstTimeOpen
-      ? [...followingTopics, {separator: true}, ...unfollowingTopics]
-      : [...newMapFollowedTopics, {separator: true}, ...newMapUnfollowedTopics];
+      ? [{separator: true}, ...unfollowingTopics]
+      : [{separator: true}, ...newMapUnfollowedTopics];
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for constructing the `firstData` and `data` arrays has been modified. This change appears to be correct assuming that `newMapFollowedTopics` and `newMapUnfollowedTopics` are correctly populated elsewhere in the code.
```diff
- const data = isFirstTimeOpen
-  ? [...followingTopics, {separator: true}, ...unfollowingTopics]
-  : [...newMapFollowedTopics, {separator: true}, ...newMapUnfollowedTopics];
+ const firstData = isFirstTimeOpen ? followingTopics : newMapFollowedTopics;
+ const data = isFirstTimeOpen
+   ? [{separator: true}, ...unfollowingTopics]
+   : [{separator: true}, ...newMapUnfollowedTopics];
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->